### PR TITLE
Address review comments related to the keyboard shortcuts refactor:

### PR DESCRIPTION
**Pull request by youknowriad**
Address review comments related to the keyboard shortcuts refactor:

https://github.com/WordPress/gutenberg/pull/19320#discussion_r362597008
https://github.com/WordPress/gutenberg/pull/19318/files#r362599806

### CODE REVIEW

File: docs/designers-developers/developers/data/data-core-keyboard-shortcuts.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ _Parameters_
 
 -   _state_ `Object`: Global state.
 -   _name_ `string`: Shortcut name.
--   _representation_ `string`: Type of reprensentation. (display, raw, ariaLabel )
+-   _representation_ `null`: Type of representation (display, raw, ariaLabel).
```
Comment (aduth) [inline docs/designers-developers/developers/data/data-core-keyboard-shortcuts.md]: Related: #18045

File: packages/keyboard-shortcuts/src/store/actions.js [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
+
 /**
  * Keyboard key combination.
  *
  * @typedef {Object} WPShortcutKeyCombination
  *
- * @property {string}  character Character.
- * @property {string} [modifier]  Modifier.
+ * @property {string}                      character Character.
+ * @property {WPKeycodeModifier|undefined} modifier  Modifier.
```
Comment (aduth) [inline packages/keyboard-shortcuts/src/store/actions.js]: This change was a bit unfortunate, I found. The previous `[modifier]` optional property syntax should be fine, but for some reason the TypeScript tooling does not pick up on it being an issue unless it was explicitly a union type with `undefined`, even though it detects it as optional.

It only (correctly) flags an error when trying to use a modifier method on `WPKeycodeHandlerByModifier` when it's typed using the undefined union:

undefined-union

**Review by aduth — APPROVED** [review]

File: packages/keyboard-shortcuts/src/store/selectors.js [hunk 3]
```diff
@@ -19,6 +19,15 @@ import { displayShortcut, shortcutAriaLabel, rawShortcut } from '@wordpress/keyc
  */
 const EMPTY_ARRAY = [];
 
+/**
+ * {{[string]:Function}} Shortcut Formatting Methods.
```
[the remark this answers is not in the document]
Comment (aduth) [reply packages/keyboard-shortcuts/src/store/selectors.js]: > we could use `Object` type combined with `@property` tags.

In implementing this, I've found that using the `{Object}` type in TypeScript has a high likelihood to trigger the `any` type fallback unintentionally, even when complemented by `@property` tags or types for the keys (see also [this syntax](https://github.com/WordPress/gutenberg/pull/19395/files#diff-d066abbb51854de985e2e0c1dd102821R35) vs. `Object<M in WPKeycodeModifier,Function>`).

We probably need to work through this a bit more. Not sure if it's a shortcoming of TypeScript, or just an accurate reflection of the "Object" type allowing for more keys than those explicitly defined (e.g. prototype members).

**Comment by youknowriad** [comment]
Thanks for the updates Andrew

**Pull request merged**